src/devices/acurite.c

```diff
@@ -662,6 +731,106 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+                    "channel",          NULL,       DATA_STRING, &channel_str,
+                    "sequence_num",     NULL,       DATA_INT,    sequence_num,
+                    "battery_ok",       NULL,       DATA_INT,    !battery_low,
+                    "message_type",     NULL,       DATA_INT,    message_type,
+                    "wind_avg_mi_h",    "Wind Speed", DATA_FORMAT, "%.1f mi/h", DATA_DOUBLE, wind_speed_mph,
+                    NULL);
+            /* clang-format on */
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_TEMP_HUMIDITY) {
+                // Wind speed, temperature and humidity
+
+                // range -40 to 160 F
+                // FIXME: are there really 13 bits? use 11 for now.
+                int temp_raw = (bb[4] & 0x0F) << 7 | (bb[5] & 0x7F);
+                tempf = (temp_raw - 400) * 0.1;
+
+                humidity = (bb[6] & 0x7f); // 1-99 %rH
+
+                /* clang-format off */
+                data = data_append(data,
+                        "temperature_F",    "temperature",  DATA_FORMAT,    "%.1f F",       DATA_DOUBLE, tempf,
+                        "humidity",         NULL,           DATA_FORMAT,    "%u %%",        DATA_INT,    humidity,
+                        NULL);
+                /* clang-format on */
+            }
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_RAINFALL) {
+                // Wind speed, wind direction, and rain fall
+                wind_dir = ((bb[4] & 0x1f) << 5) | ((bb[5] & 0x7c) >> 2);
+
+                // range: 0 to 99.99 in, 0.01 inch increments, accumulated
+                // FIXME: are there really only 7 bits? use 9 for now.
+                raincounter = ((bb[5] & 0x03) << 7) | (bb[6] & 0x7F);
+
+                /* clang-format off */
+                data = data_append(data,
+                        "wind_dir_deg",     NULL,           DATA_FORMAT,    "%.1f",         DATA_DOUBLE, wind_dir,
+                        "rain_in",          "Rainfall Accumulation", DATA_FORMAT, "%.2f in", DATA_DOUBLE, raincounter * 0.01f,
+                        NULL);
+                /* clang-format on */
+            }
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_UV_LUX) {
+                // Wind speed, UV Index, Light Intensity, Lightning?
+                int uv = (bb[4] & 0x3f);
+                int lux = ((bb[5] & 0x7f) << 7) | (bb[6] & 0x7F);
+
+                /* clang-format off */
+                data = data_append(data,
+                        "uv",               NULL,           DATA_INT, uv,
+                        "lux",              NULL,           DATA_INT, lux * 10,
+                        NULL);
+                /* clang-format on */
+            }
+
+            int byte8  = (bb[7] & 0x7f);
+            int byte9  = (bb[8] & 0x7f);
+            int byte89 = ((bb[7] & 0x7f) << 7) | (bb[8] & 0x7F);
+
+            /* clang-format off */
+            data = data_append(data,
```

Comment:
I think this  approach is useful to get things moving for the fields that are well understood but make it easier to find/process/reprocess data that might not be decoded correctly.

However, I suggest dumping *all* the bytes of the message as one field: "raw_msg" until confidence is very high that the message is fully decoded accurately.  It makes it easy to save (or ignore) the raw message with the associated timestamp.